File: backend/schemas/financial_aims.py
from pydantic import BaseModel
from typing import Optional

class FinancialAimBase(BaseModel):
    title: str
    description: Optional[str] = None
    target_amount: float
    current_amount: float = 0.0

class FinancialAimCreate(FinancialAimBase):
    pass

class FinancialAimUpdate(BaseModel):
    title: Optional[str]
    description: Optional[str]
    target_amount: Optional[float]
    current_amount: Optional[float]

class FinancialAimResponse(FinancialAimBase):
    id: int
    user_id: int

    class Config:
        orm_mode = True

File: frontend/src/types/financialAims.ts
export interface FinancialAim {
    id: number;
    user_id: number;
    title: string;
    description?: string | null;
    target_amount: number;
    current_amount: number;
}

export type FinancialAimCreate = Omit<FinancialAim, 'id' | 'user_id'>;
export type FinancialAimUpdate = Partial<FinancialAimCreate>;

export type FinancialTx = {
    amount: number,
    transaction_type: string,
    id: number,
    created_at: string,
    bank_account_id: number
}